### DISCUSSION TITLE
webhook payloads for data type are different

### DISCUSSION THREAD

**Discussion by maxknee** [community/community#48987] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

When inspecting webhook payloads from repositories, I've noticed that for push events, `repository.created_at` is an int and for all other events, is a timestamp.

Is there a reason why they differ? Can we normalize them? Could it be either a timestamp or an int since the repository object should be identical across event types.

Happy to discuss further

**Comment by wbaker23** [comment]
I see the same behavior for our webhooks.
https://github.com/orgs/community/discussions/148957 is a newer discussion thread related to this issue.

**Comment by Devanand-official** [comment]
Good catch — you’re right, the repository.created_at field behaves inconsistently across different webhook event types.

For push events, it’s returned as a Unix timestamp (int).

For other events, it’s returned as an ISO 8601 timestamp (string).

This mismatch is a known inconsistency in the GitHub API’s webhook schema. Currently, GitHub hasn’t standardized this field across all event types.

As a workaround, you can normalize it in your webhook handler — for example, convert integers to ISO timestamps before processing.

You can also open an issue on the GitHub REST API discussions to track or upvote schema normalization.
